pkg/cloudprovider/gcp/object_store.go

```diff
@@ -60,28 +62,45 @@ type ObjectStore struct {
 	googleAccessID string
 	privateKey     []byte
 	bucketWriter   bucketWriter
+	iamSvc         *iamcredentials.Service
 }
 
 func NewObjectStore(logger logrus.FieldLogger) *ObjectStore {
 	return &ObjectStore{log: logger}
 }
 
 func (o *ObjectStore) Init(config map[string]string) error {
-	if err := cloudprovider.ValidateObjectStoreConfigKeys(config); err != nil {
+	if err := cloudprovider.ValidateObjectStoreConfigKeys(config, serviceAccountConfig); err != nil {
 		return err
 	}
+	/* Find default token source to extract the GoogleAccessID */
+	ctx := context.Background()
+	creds, err := google.FindDefaultCredentials(ctx)
```

Comment:
I think it makes sense to first check for an `err != nil` here and return it if an error's gotten, and only after that proceed to checking the `JSON` field